### PR TITLE
feat(SetTheory/Surreal/Basic): add some simp lemmas for Surreal.mk

### PR DESCRIPTION
**Pull request by hwatheod**
add some simp lemmas for Surreal.mk

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Surreal/Basic.lean [hunk 1]
```diff
@@ -301,6 +301,15 @@ instance : One Surreal :=
...
   ⟨0⟩
 
+@[simp]
+lemma mk_eq_mk {x y : PGame.{u}} {hx hy} : mk x hx = mk y hy ↔ x ≈ y := by
+  rw [mk, mk, Quotient.eq]; rfl
+
+@[simp]
+lemma mk_eq_zero {x : PGame.{u}} {hx} : mk x hx = 0 ↔ x ≈ 0 := by
+  change mk x hx = mk 0 numeric_zero ↔ x ≈ 0
+  rw [mk_eq_mk]
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Surreal/Basic.lean:311]:
```suggestion
@[simp]
lemma mk_eq_mk {x y : PGame.{u}} {hx hy} : mk x hx = mk y hy ↔ x ≈ y := Quotient.eq

@[simp]
lemma mk_eq_zero {x : PGame.{u}} {hx} : mk x hx = 0 ↔ x ≈ 0 := Quotient.eq
```

**Review by eric-wieser — COMMENTED** [review]
I'm not sure these should be `@[simp]`; [`QuotientAddGroup.mk'_eq_mk'`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/QuotientGroup.html#QuotientAddGroup.mk'_eq_mk') is morally very similar, and is also not `simp`.

File: Mathlib/SetTheory/Surreal/Basic.lean (hunk 1 above)
Comment (hwatheod) [reply Mathlib/SetTheory/Surreal/Basic.lean:311]: Are there guidelines for when something should be `@[simp]` ? If I don't tag these as `@[simp]`, then I need to list them explicitly within `simp [...]` when I use them. In my current work, there is just one use of each of these two lemmas, and there may a few more later, but I don't anticipate a huge number.

Comment (hwatheod) [reply Mathlib/SetTheory/Surreal/Basic.lean:311]: The `mk_le_mk` a few lines below these changes is currently `@[simp]`.

Comment (hwatheod) [reply Mathlib/SetTheory/Surreal/Basic.lean]: Keeping the `@[simp]` for now.

**Comment by hwatheod** [comment]
Closing this PR for now as further work as made it less and less clear that we want these to be `@[simp]` lemmas. May re-open in the future.

**Comment by eric-wieser** [comment]
Even if the value of `simo` is dubious, having these lemmas without simp seems like a good thing to me, and I'd be happy to merge that version

**Comment by hwatheod** [comment]
Re-opening with `simp` removed and some additional lemmas added, which are needed for the division proof

**Comment by hwatheod** [comment]
Reopning

**Pull request closed without merging**